pkg/registry/graph.go

```diff
@@ -264,6 +264,9 @@ func NewGraph(stepsByName ReferenceMap, chainsByName ChainMap, workflowsByName W
 		nodesByName[name] = node
 		for _, step := range chain {
 			if step.Reference != nil {
+				if referenceNodes[*step.Reference] == nil {
```

Comment:
nit: idiomatic Go for map key existence is `if _, exists := myMap[key]; !exists {`

Reply:
Fixed